module/Application/src/Application/ProjectUrlValidator.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+
+namespace Application;
+
+
+use Zend\Validator\Exception;
+use Zend\Validator\ValidatorInterface;
+
+class ProjectUrlValidator implements ValidatorInterface {
+    protected $errors = [];
+    
+    /**
+     * Returns true if and only if $value meets the validation requirements
+     *
+     * If $value fails validation, then this method returns false, and
+     * getMessages() will return an array of messages that explain why the
+     * validation failed.
+     *
+     * @param  mixed $value
+     * @return bool
+     * @throws Exception\RuntimeException If validation of $value is impossible
+     */
+    public function isValid($value) {
+        if (!filter_var($value, FILTER_VALIDATE_URL)) {
+            $this->errors['invalid-url'] = "Invalid URL";
+        }
+        
+        if (strpos($value, '/..') !== false) {
```

Comment:
Path components can start with two dots. There is a path traversal issue only if the entire component two dots.